### PR TITLE
[FLINK-16486][python][docs] Add documentation for vectorized Python UDF

### PR DESCRIPTION
**Pull request by dianfu**
## What is the purpose of the change

*This pull request add documentation for vectorized Python UDF.*

## Brief change log

  - *Add documentation `vectorized_python_udfs.md` for vectorized Python UDF*

## Verifying this change

This change is a documentation rework without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: docs/dev/table/python/vectorized_python_udfs.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+Vectorized user-defined functions are functions which are executed by transferring a batch of elements between JVM and Python VM in Arrow columnar format.
+It shares the similar way as the [non-vectorized user-defined functions]({{ site.baseurl }}/dev/table/python/python_udfs.html) on how to
+define vectorized user-defined functions. Users only need to add an extra parameter udf_type="pandas" in the decorator `udf` to mark it as a vectorized user-defined function.
```
Comment (hequn8128) [inline docs/dev/table/python/vectorized_python_udfs.md:27]: change to \`udf_type="pandas"\`

File: docs/dev/table/python/vectorized_python_udfs.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+Vectorized user-defined functions are functions which are executed by transferring a batch of elements between JVM and Python VM in Arrow columnar format.
```
Comment (hequn8128) [inline docs/dev/table/python/vectorized_python_udfs.md:25]: Also add some descriptions about the advantages of the vectorized udfs over regular udfs?

**Comment by dianfu** [comment]
@hequn8128 Thanks for the review. Updated the PR.

File: docs/dev/table/python/vectorized_python_udfs.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+Users only need to add an extra parameter `udf_type="pandas"` in the decorator `udf` to mark it as a vectorized user-defined function.
+
+**NOTE:** Python UDF execution requires Python version (3.5, 3.6 or 3.7) with PyFlink installed. It's required on both the client side and the cluster side. 
+
+* This will be replaced by the TOC
+{:toc}
+
+## Vectorized Scalar Functions
+
+Vectorized Python scalar functions take pandas.Series as the inputs and return a pandas.Series of the same length as the output.
```
Comment (hequn8128) [inline docs/dev/table/python/vectorized_python_udfs.md:39]: to \`pandas.Series\`. Also note for other places.

**Comment by dianfu** [comment]
@hequn8128 Thanks for the review. Updated the PR.

**Comment by hequn8128** [comment]
@dianfu Thanks a lot for the update. Merging...

**Pull request merged**